geo/src/algorithm/dimensions.rs

```diff
@@ -0,0 +1,452 @@
+    /// Dimension of a line or curve
+    OneDimensional,
+    /// Dimension of a surface
+    TwoDimensional,
+}
+
+/// Operate on the dimensionality of geometries.
+pub trait HasDimensions {
+    /// Some geometries, like a `MultiPoint`, can have zero coordinates - we call these `empty`.
+    ///
+    /// Types like `Point` and `Rect`, which have at least one coordinate by construction, can
+    /// never be considered empty.
+    /// ```
+    /// use geo_types::{Point, Coordinate, LineString};
+    /// use geo::algorithm::dimensions::HasDimensions;
+    ///
+    /// let line_string = LineString(vec![
+    ///     Coordinate { x: 0., y: 0. },
+    ///     Coordinate { x: 10., y: 0. },
+    /// ]);
+    /// assert!(!line_string.is_empty());
+    ///
+    /// let empty_line_string: LineString<f64> = LineString(vec![]);
+    /// assert!(empty_line_string.is_empty());
+    ///
+    /// let point = Point::new(0.0, 0.0);
+    /// assert!(!point.is_empty());
+    /// ```
+    fn is_empty(&self) -> bool;
+
+    /// The dimensions of some geometries are fixed, e.g. a Point always has 0 dimensions. However
+    /// for others, the dimensionality depends on the specific geometry instance - for example
+    /// typical `Rect`s are 2-dimensional, but it's possible to create degenerate `Rect`s which
+    /// have either 1 or 0 dimensions.
+    ///
+    /// ## Examples
+    ///
+    /// ```
+    /// use geo_types::{GeometryCollection, Rect, Point};
+    /// use geo::algorithm::dimensions::{Dimensions, HasDimensions};
+    ///
+    /// // normal rectangle
+    /// let rect = Rect::new((0.0, 0.0), (10.0, 10.0));
+    /// assert_eq!(Dimensions::TwoDimensional, rect.get_dimensions());
+    ///
+    /// // "rectangle" with zero height degenerates to a line
+    /// let degenerate_line_rect = Rect::new((0.0, 10.0), (10.0, 10.0));
+    /// assert_eq!(Dimensions::OneDimensional, degenerate_line_rect.get_dimensions());
+    ///
+    /// // "rectangle" with zero height and zero width degenerates to a point
+    /// let degenerate_point_rect = Rect::new((10.0, 10.0), (10.0, 10.0));
+    /// assert_eq!(Dimensions::ZeroDimensional, degenerate_point_rect.get_dimensions());
+    ///
+    /// // collections inherit the greatest dimensionality of their elements
+    /// let geometry_collection = GeometryCollection(vec![degenerate_line_rect.into(), degenerate_point_rect.into()]);
+    /// assert_eq!(Dimensions::OneDimensional, geometry_collection.get_dimensions());
+    ///
+    /// let point = Point::new(10.0, 10.0);
+    /// assert_eq!(Dimensions::ZeroDimensional, point.get_dimensions());
+    ///
+    /// // An `Empty` dimensionality is distinct from, and less than, being 0-dimensional
+    /// let empty_collection = GeometryCollection::<f32>(vec![]);
+    /// assert_eq!(Dimensions::Empty, empty_collection.get_dimensions());
+    /// assert!(empty_collection.get_dimensions() < point.get_dimensions());
+    /// ```
+    fn get_dimensions(&self) -> Dimensions;
```

Comment:
thoughts on removing the `get_` prefix? getters in rust tend to not have 'get'

https://rust-lang.github.io/api-guidelines/naming.html#getter-names-follow-rust-convention-c-getter

Reply:
Love it! Honestly I assumed you couldn't have a method and property with the same name. 😅